### PR TITLE
[BugFix] Fix parse comment incompatible bug

### LINKED ISSUE #7479: create routine load with comments will fail after FE restart
### Steps to reproduce the behavior (Required)
Use dbeaver to send the following sql to create a routine job, then restart current FE master.
```sql
--- xxx
 CREATE ROUTINE LOAD job ON agg_table
 COLUMNS TERMINATED BY ",",
 COLUMNS (user_id, web_url, pv)
 PROPERTIES
 (
   "desired_concurrent_number" = "6",
   "max_batch_interval" = "10",
   "max_error_number" = "1000"
 )
 FROM KAFKA
 (
   "kafka_broker_list" = "172.26.92.155:9092",
   "kafka_topic" = "test",
   "property.kafka_default_offsets" = "OFFSET_BEGINNING",
   "kafka_partitions" = "0"
 );
```

### Expected behavior (Required)
master failover success.

### Real behavior (Required)
new master exit with error msg: Invalid statement index: 1, size: 1

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
main

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7479

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/SqlParser.java|8|8|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/SqlParser.java [hunk 1]
```diff
@@ -88,7 +90,28 @@ public static StatementBase parseWithOldParser(String originStmt, long sqlMode,
...
+     * Because originSql is stored in the old version of the materialized view, it is the index stored in the old version.
+     * */
+    static String regexSimpleComment = "--.*\n";
+    static Pattern psm = Pattern.compile(regexSimpleComment);
+
+    static String regexBracketComment = "/\\* .*? \\*/";
+    static Pattern pbc = Pattern.compile(regexBracketComment);
+
     private static List<String> splitSQL(String sql) {
+        Matcher m = psm.matcher(sql);
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/SqlParser.java:110]: Would it make the comment disappear in audit log and other logs?

Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/SqlParser.java:110]: I don't think so. The original SQL is stored in logs as well as XXStmt class.

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeSingleTest.java [hunk 2]
```diff
@@ -494,6 +494,12 @@ public void testSqlSplit() {
 
         list = SqlParser.parse("select * from t1 where a1 = 'x\"x;asf';", 0);
         Assert.assertEquals(1, list.size());
+
+        list = SqlParser.parse("-- xxx;\nselect 1;", 0);
+        Assert.assertEquals(1, list.size());
```
Comment (padmejin) [inline fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeSingleTest.java:499]: I suggest you add an assert on the content of the result string. 
```
Assert.assertEquals("select 1;", list.get(0));
```

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/SqlParser.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/SqlParser.java:110]: no, only for parse

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/SqlParser.java|8|8|100.00%|

**Review by padmejin — APPROVED** [review]
LGTM

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**